ppl/zqd/temporal/spaceworkflow.go

```diff
@@ -0,0 +1,100 @@
+package temporal
+
+import (
+	"context"
+	"time"
+
+	"github.com/brimsec/zq/api"
+	"github.com/brimsec/zq/ppl/zqd/apiserver"
+	"go.temporal.io/sdk/activity"
+	"go.temporal.io/sdk/worker"
+	"go.temporal.io/sdk/workflow"
+)
+
+var config Config
+var manager *apiserver.Manager
+
+func InitSpaceWorkflow(c Config, m *apiserver.Manager, r worker.Registry) {
+	config = c
+	manager = m
+	r.RegisterWorkflow(spaceWorkflow)
+	r.RegisterActivity(spaceCompactActivity)
+	r.RegisterActivity(spacePurgeActivity)
+}
+
+const writeSignal = "write"
+
+func spaceWorkflow(ctx workflow.Context) error {
+	logger := workflow.GetLogger(ctx)
+	logger.Info("Workflow started")
+	signalCh := workflow.GetSignalChannel(ctx, writeSignal)
+	for {
+		logger.Info("Workflow waiting for signal")
+		var signalVal string
+		signalCh.Receive(ctx, &signalVal)
+
+		logger.Info("Workflow waiting for timer or signal")
+		s := workflow.NewSelector(ctx)
+		timerCtx, cancel := workflow.WithCancel(ctx)
+		timerFuture := workflow.NewTimer(timerCtx, config.SpaceCompactDelay)
+		var err error
+		var executedActivity bool
+		s.AddFuture(timerFuture, func(f workflow.Future) {
+			logger.Info("Workflow timer expired")
+			ctx := workflow.WithActivityOptions(ctx, workflow.ActivityOptions{
+				StartToCloseTimeout: 12 * time.Hour,
```

Comment:
The temporal go docs say that both `StartToCloseTimeout` (which is here) and `ScheduleToStartTimeout` are mandatory, but this is clearly working without ScheduleToStartTimeout. Can you see if its really mandatory, in case not having a value leads to unexplained errors later?

Reply:
I think this is OK. The Go SDK docs at https://pkg.go.dev/go.temporal.io/sdk@v1.4.1/ look like they're out of sync with the underlying implementation at https://pkg.go.dev/go.temporal.io/api@v1.4.0/command/v1#ScheduleActivityTaskCommandAttributes, which says that either `ScheduleToCloseTimeout` or `StartToCloseTimeout` must be set.